src/main/java/com/nylas/NeuralSignatureContact.java

```diff
@@ -0,0 +1,133 @@
+package com.nylas;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class NeuralSignatureContact {
+	private List<String> job_titles;
+	private List<Links> links;
+	private List<Names> names;
+	private List<String> phone_numbers;
+	private List<String> emails;
+
+	public List<String> getJobTitles() {
+		return job_titles;
+	}
+
+	public void setJobTitles(List<String> jobTitles) {
+		this.job_titles = jobTitles;
+	}
+
+	public List<Links> getLinks() {
+		return links;
+	}
+
+	public void setLinks(List<Links> links) {
+		this.links = links;
+	}
+
+	public List<Names> getNames() {
+		return names;
+	}
+
+	public void setNames(List<Names> names) {
+		this.names = names;
+	}
+
+	public List<String> getPhoneNumbers() {
+		return phone_numbers;
+	}
+
+	public void setPhoneNumbers(List<String> phoneNumbers) {
+		this.phone_numbers = phoneNumbers;
+	}
+
+	public List<String> getEmails() {
+		return emails;
+	}
+
+	public void setEmails(List<String> emails) {
+		this.emails = emails;
+	}
+
+	public Contact toContactObject() {
+		Contact contact = new Contact();
+
+		if(names != null) {
+			contact.setGivenName(names.get(0).getFirstName());
+			contact.setSurname(names.get(0).getLastName());
+		}
+		if(job_titles != null) {
+			contact.setJobTitle(job_titles.get(0));
+		}
+		if(emails != null) {
+			List<Contact.Email> emailList = new ArrayList<>();
+			for(String email : emails) {
+				emailList.add(new Contact.Email("personal", email));
+			}
+			contact.setEmails(emailList);
+		}
+		if(phone_numbers != null) {
+			List<Contact.PhoneNumber> phoneList = new ArrayList<>();
+			for(String number : phone_numbers) {
+				phoneList.add(new Contact.PhoneNumber("mobile", number));
+			}
+			contact.setPhoneNumbers(phoneList);
+		}
+		if(links != null) {
+			List<Contact.WebPage> webPageList = new ArrayList<>();
+			for(Links link : links) {
+				String description = link.getDescription() != null && !link.getDescription().isEmpty() ?
+						link.getDescription() : "homepage";
+				webPageList.add(new Contact.WebPage(description, link.getUrl()));
+			}
+			contact.setWebPages(webPageList);
+		}
+
+		return contact;
+	}
+
+	@Override
+	public String toString() {
+		return String.format("NeuralSignatureContact [job_titles=%s, links=%s, names=%s, phone_numbers=%s, emails=%s]",
+				job_titles, links, names, phone_numbers, emails);
+	}
+}
+
+class Names {
```

Comment:
Good call out on the nesting of classes, that's the better approach. Also regarding renaming the class name, is there a reason why you're against just `Names`? I only ask because I'm trying to keep variable naming consistent across all SDKs and I've already used `Names` everywhere else.

Reply:
I suggested renaming this class to a singular form to indicate that it is holding data for a single contact, to contrast it with the list of them above.  But I don't feel strongly, and consistency with other SDKs is a good counter argument.  Fine with either way.

Reply:
You make a great point actually. It should be fine to cascade the changes in the other SDKs anyways post-release since the users never directly interact with subclasses like these. I'll make the change, good catch!

Reply:
This class name will be visible to users of the Java SDK, though I'm not familiar enough with the other SDKs to say.  Technically it would be an incompatible api change for this Java SDK to change it after a release, though a pretty minor one.